docs/runtime_api.md

```diff
@@ -69,12 +71,46 @@ new User({
 });
 ```
 
-Note that all fields in the initializer object are optional, and if not
-provided, the default value for the field is used.
+The initializer object accepted by all message constructors is of type
+[`PartialMessage<T>`](src-partial-message) where `T` is your message type. So
+in the above example, the initializer object is of type `PartialMessage<User>`.
+`PartialMessage` is similar to the TypeScript built-in type `Partial`, but works
+recursively. For more details, see the below section on [Advanced TypeScript types](#advanced-typescript-types).
+
+If you need to define the initializer object independent of the constructor,
+then be sure to use a type assertion, otherwise you may see unexpected compile
+errors with `oneof` fields. In TypeScript 4.9 and above, it is recommended to
+use `satisfies`.
+
+```typescript
+const obj = {
+  firstName: "Homer",
+  active: true,
+  manager: {
+    lastName: "Burns",
+  },
+} satisfies PartialMessage<User>;
+
+const user = new User(obj);
+```
+
+Otherwise, use the `as` keyword:
+
+```typescript
+const obj = {
+  firstName: "Homer",
+  active: true,
+  manager: {
+    lastName: "Burns",
+  },
+} as PartialMessage<User>;
+
+const user = new User(obj);
```

Comment:
We should not recommend to use `as`. This is that the `case` property of a literal is typed as `string`. `as const` is the better solution here:

```suggestion
If you are not using TypeScript 4.9 yet, use `as const` instead of `satisfies PartialMessage<User>`.
```